api/metric/observer.go

```diff
@@ -33,3 +42,185 @@ type Int64Observer struct {
 type Float64Observer struct {
 	asyncInstrument
 }
+
+// BatchObserver represents an Observer callback that can report
+// observations for multiple instruments.
+type BatchObserver struct {
+	meter  Meter
+	runner AsyncBatchRunner
+}
+
+// Int64ObserverResult is passed to an observer callback to capture
+// observations for one asynchronous integer metric instrument.
+type Int64ObserverResult struct {
+	instrument AsyncImpl
+	function   func([]core.KeyValue, Observation)
+}
+
+// Float64ObserverResult is passed to an observer callback to capture
+// observations for one asynchronous floating point metric instrument.
+type Float64ObserverResult struct {
+	instrument AsyncImpl
+	function   func([]core.KeyValue, Observation)
+}
+
+// BatchObserverResult is passed to a batch observer callback to
+// capture observations for multiple asynchronous instruments.
+type BatchObserverResult struct {
+	function func([]core.KeyValue, []Observation)
+}
+
+// AsyncRunner is an interface implemented by all Observer callbacks.
+// observers run.
+type AsyncRunner interface {
+	anyRunner()
```

Comment:
A comment is needed to explain how this is a mechanism to keep runners separated across different implementations.

Reply:
Fixed.